### PR TITLE
Added Safer Teleport options.

### PR DESCRIPTION
**Pull request by inburst21**
Added checks to find the distance to the next PokeStop. If the distance is over the distance you set in the config file defaulted at 35m then it will keep teleporting your specified distance. Greatly reduces softbans.

### CODE REVIEW

File: PoGo.PokeMobBot.Logic/Navigation.cs [hunk 1]
```diff
@@ -169,16 +172,51 @@ public Navigation(Client client)
...
+
+
+
+                    var requestSendDateTime = DateTime.Now;
+                    var result =
+                        await
+                            _client.Player.UpdatePlayerLocation(waypoint.Latitude, waypoint.Longitude,
+                                _client.Settings.DefaultAltitude);
+                    currentDistanceToTarget = LocationUtils.CalculateDistanceInMeters(sourceLocation, targetLocation);
+                    Logging.Logger.Write("We are teleporting " + maxDistance + " meters closer to the target. We are now " + currentDistanceToTarget + " away");
```
Comment (DurtyFree) [inline PoGo.PokeMobBot.Logic/Navigation.cs]: Please use the EventDispatcher and make this a translation string

**Comment by DurtyFree** [comment]
Need opinions and testers for this!:)

File: PoGo.PokeMobBot.Logic/Navigation.cs (hunk 1 above)
Comment (inburst21) [inline PoGo.PokeMobBot.Logic/Navigation.cs]: I really meant to comment that out. Its console spam really. I seem to be getting a solid 100k exp an hour with these settings without sniping. Also Im not sure how to edit the file once sent to the request. Thats why I have closed my previous 3

**Comment by nwg5817** [comment]
Can't compile the latest build unfortunately. Will test this asap. Definitely has promise!

**Comment by nothingtoseeheremates** [comment]
Yeah i think so too, thanks for adding that, had asked earlier between than and other option, but if possible create a minimum distance like nwg5817 said, i think it would be better to make it > 100 meters before it staggers teleports (and do the 100 meteres instantly first and then the small teleports for things further away than 100m)

Cheers

**Comment by CodeOfTheRing** [comment]
Ok also to add to this. After much testing I run around an area that has a couple spots where it goes from 300m-900m then back down to 100ms where you don't get softbanned much. So this safe teleporting doesn't reduce the softbanning at the 300m-600m-900m mark. Here is my suggestion. Slow down the safe teleporting by integrating a search for pokemon at each leap. Option safeteleport search for pokemon. Also add a settings option for the time between leaps and you could truly customize your routes and maximize xp. We really do require minimum distance required either way, because I'd like to test whether 5m jumps would stop softbanning for long trips, it's just too costly for the entire route to use that low a leap. I stand behind my original comments that this does a great job and its a worthy addon. We just need a little more tweaking to reach optimum performance.

**Comment by inburst21** [comment]
I have a much better one coming out later today. It learns as it goes. It will become soft ban proofish. If they ever update their code mine will do so automatically. Gotta love AI

**Comment by forReason** [comment]
like it!

**Comment by DurtyFree** [comment]
Is this tested?
@inburst21 Please fix the outstanding merge conflicts

**Comment by inburst21** [comment]
@DurtyFree. I still have to fix the waitTime not saving to the teleai .json not sure why it's not

**Comment by OverCraftX** [comment]
A good fix its just to launch the sniper procedure if the pokestop its far away. After the sniper come back it will spin. it just need to wait a bit time before spinning the pokestop after the jump. It doesn't matter how many times you spin it or how many in between jumps you make. Its only a time delay for futher jumps, usefull for reciclyng, renaming, snipping etc..

Making more jumps for a far away pokestops just add a delay per jump, and that's why it seems to be working, but it only decrease the exp/hour. My solution its a bit better upgrade.

**Comment by inburst21** [comment]
@OverCraftX I have done a few upgrades since this was posted. It now does one jump and then determines how long is for optimizing your exp with no softbans. If you received a softban from a jump it will detect that and automatically add a delay for that distance group

**Pull request closed without merging**